framework/MCMC/Metropolis.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Markov Chain Monte Carlo
+  This base class defines the principle methods required for MCMC
+
+  Created on June 26, 2020
+  @author: wangc
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import numpy as np
+import copy
+import abc
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+
+from .MCMC import MCMC
+from utils import utils,randomUtils,InputData, InputTypes
+#Internal Modules End--------------------------------------------------------------------------------
+
+class Metropolis(MCMC):
+  """
+    Metropolis Sampler
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(Metropolis, cls).getInputSpecification()
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Default Constructor that will initialize member variables with reasonable
+      defaults or empty lists/dictionaries where applicable.
+      @ In, None
+      @ Out, None
+    """
+    MCMC.__init__(self)
+    self._localReady = True # True if the submitted job finished
+    self._currentRlz = None # dict stores the current realizations, i.e. {var: val}
+
+  def handleInput(self, paramInput):
+    """
+      Read input specs
+      @ In, paramInput, InputData.ParameterInput, parameter specs interpreted
+      @ Out, None
+    """
+    MCMC.handleInput(self, paramInput)
+
+  def initialize(self, externalSeeding=None, solutionExport=None):
+    """
+      This function should be called every time a clean MCMC is needed. Called before takeAstep in <Step>
+      @ In, externalSeeding, int, optional, external seed
+      @ In, solutionExport, DataObject, optional, a PointSet to hold the solution
+      @ Out, None
+    """
+    MCMC.initialize(self, externalSeeding=externalSeeding, solutionExport=solutionExport)
```

Comment:
this method is not needed if you just call the baseclass method...you can just remove it